crates/diesel_models/src/query/payment_method.rs

```diff
@@ -173,3 +187,108 @@ impl PaymentMethod {
         }
     }
 }
+
+#[cfg(all(feature = "v2", feature = "payment_methods_v2"))]
+impl PaymentMethod {
+    pub async fn find_by_id(conn: &PgPooledConn, id: &str) -> StorageResult<Self> {
+        generics::generic_find_one::<<Self as HasTable>::Table, _, _>(conn, pm_id.eq(id.to_owned()))
+            .await
+    }
+
+    pub async fn find_by_customer_id_merchant_id(
+        conn: &PgPooledConn,
+        customer_id: &common_utils::id_type::CustomerId,
+        merchant_id: &common_utils::id_type::MerchantId,
+        limit: Option<i64>,
+    ) -> StorageResult<Vec<Self>> {
+        generics::generic_filter::<<Self as HasTable>::Table, _, _, _>(
+            conn,
+            dsl::customer_id
+                .eq(customer_id.to_owned())
+                .and(dsl::merchant_id.eq(merchant_id.to_owned())),
+            limit,
+            None,
+            Some(dsl::last_used_at.desc()),
+        )
+        .await
+    }
+
+    pub async fn get_count_by_customer_id_merchant_id_status(
```

Comment:
If this is only for OLAP add feature gate

Reply:
This query was introduced for default payment method flow, we can decide later on if this is needed for v2, have removed it for now.